.github/workflows/build-multiarch.yml

```diff
@@ -12,7 +12,22 @@ on:
       - "examples/**"
 
 jobs:
+  setup:
+    runs-on: ubuntu-20.04
+    outputs:
+      # output the generated rcon_password to be used as build arg later.
+      rcon_password: ${{ steps.passwdgen.outputs.password }}
+    steps:
+      # the setup for the rcon password is done outside of the matrix, so each 
+      # build uses the same generated password in all the image variants.
+      - name: Create RCON Password
+        uses: licenseware/generate-password-and-hash@v1
+        id: passwdgen
+        with: 
+          length: 20
   build:
```

Comment:
From a CI perspective, I like the idea of this setup stage. However, I was expecting a simpler approach where the build job just included this step inline, which has the nice side effect of "randomizing" the passwords a bit more, one per image variant.

Reply:
One per image variant is fine too, i just thought to do it once and reuse that, but either way works, since the password doesnt really matter except for the tests.